### DISCUSSION TITLE
0.6.1 Nuget Release

### DISCUSSION THREAD

**Discussion by MattB-msft** [microsoft/PowerPlatform-DataverseServiceClient#240] [General]
Minor Bump to 0.6.x - *Should* be last bump prior to 1.0. 

- Added initial support for AD auth for OnPremise.  This will work for full framework only. 
- Added new constructor that will accept a ConnectionOptions class and a ConfigurationOptions Options Object.  This constructor is currently not fully implemented. 
  - This is prep work for a builder process and intended to unify the connection flow to a single constructor at some point in the future. 
  - The goal of this work will be to resolve the issues caused by lazy loading hosts. 
- Fixed a bug in Logger when working with Connection Strings where log events would not be correctly generated. 
- Speculative fix for GIT issue #219 and #220
- Fixed a bug where LastError was not properly reset, fixed Git #229
- Fixed Error message issue between onPrem and onLine , Fixed Git #232
- Added configuration support to allow for better memory management of very large return sets,  Git #207 
  - New AppSetting is called "MaxBufferPoolSizeOveride" - there is currently not a way to set this via code at runtime. 
  - Support for configuration of this will be added as part of the new Constructor for .net core, .net framework can use it by setting the appsetting value now.
- Fixed exceptions not being thrown from several paths during login flows. 

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/PowerPlatform-DataverseServiceClient/releases/tag/0.6.1'>0.6.1 Nuget Release</a>.</em>

**Comment by jordimontana82** [comment]
Hi @MattB-msft ,

Quick question re this version. If this is the last version before 1.0, does that mean 1.0 will likely introduce breaking changes?

**Reply by MattB-msft** [reply]
right now, no.  our goal is to take this forward. 
Things on our "to-do" list for 1.0 are to have finished the Configuration Driven Constructor which will likely add a .Connect() method to the client, and make progress on adding in builder support for DI. 
Things on our "maybe" list in this space are to provide further segregation of extensions using NS

**Reply by jordimontana82** [reply]
Great, thanks Matt!

**Comment by ciaomatteo** [comment]
Hi @MattB-msft 
I assume AD auth for OnPremises only works in conjunction with ADFS?

**Reply by MattB-msft** [reply]
No, AD auth does not require ADFS. 
OAuth on Premises does require ADFS.
